Sources/MessagingInApp/MessagingInAppConfigOptions.swift

```diff
@@ -1,2 +1,59 @@
+import CioInternalCommon
+
+/**
+ Configuration options for in-app messaging module
+
+ Example usage:
+ ```
+ // create a new instance
+ let inAppMessagingConfig = MessagingInAppConfigOptions.Factory.create(siteId: siteId, region: .US)
+ ```
+ */
+// TODO: [CDP] Update docs to after module initialization changes for clarity on its usage
 public struct MessagingInAppConfigOptions {
+    // Used to create new instance of MessagingInAppConfigOptions when the MessagingInApp module is configured.
+    // Each property of the MessagingInAppConfigOptions object can be modified by the user.
+    public enum Factory {
+        // TODO: [CDP] Rethink on how can we eliminate default value and force required properties
+        public static func create(siteId: String = "", region: Region = Region.US) -> MessagingInAppConfigOptions {
+            MessagingInAppConfigOptions(
+                siteId: siteId,
+                region: region
+            )
+        }
+    }
+    
+    public enum Keys: String {
+        case siteId
+        case region
+    }
+    
+    /// Property to store workspace Site ID set during module initialization
+    /// Immutable for lazy updates only, can be updated using initialize method
+    public private(set) var siteId: String
+    /// Property to store workspace Region set during module initialization
+    /// Immutable for lazy updates only, can be updated using initialize method
+    public private(set) var region: Region
+
+    public mutating func initialize(siteId: String, region: Region? = nil) {
+        self.siteId = siteId
+        if let region = region {
+            self.region = region
+        }
+    }
+    
+    mutating func apply(with dictionary: [String: Any]) {
+        // Each SDK config option should be able to be set from `dictionary`.
+        // If one isn't provided, use current value instead.
+        
+        // If a parameter takes more logic to calculate, perform the logic up here.
+        if let regionStringValue = dictionary[Keys.region.rawValue] as? String {
+            self.region = Region.getRegion(from: regionStringValue)
+        }
+
+        // Construct object with all required parameters. Each config option should be updated from `dictionary` only if available.
+        if let siteId = dictionary[Keys.siteId.rawValue] as? String {
+            self.siteId = siteId
+        }
+    }
```

Comment:
```suggestion
    internal static func init(from dictionary: [String: Any]) -> MessagingInAppConfigOptions? {
        guard let regionStringValue = dictionary[Keys.region.rawValue] as? String, let region = Region.getRegion(from: regionStringValue) else {
          return nil 
        }
        
        guard let siteId = dictionary[Keys.siteId.rawValue] as? String else {
          return nil 
        }
        
        return MessagingInAppConfigOptions(siteId: siteId, region: region)
    }
```

I suggest that instead of a mutating function, we take advantage of the struct constructor. 

Why? Let's say that in the future we add a new property to the `MessagingInAppConfigOptions` struct, `enable: Bool`. 

If we use a mutating function, we need to remember to go into that function and add support for this new `enable` property. If we take advantage of the struct constructor, the Swift compiler will not compile and remind us that line `return MessagingInAppConfigOptions(siteId: siteId, region: region)` is missing `enable` parameter.

Reply:
I agree with this. Initially, I wanted to go with the constructor option for same reason, but avoided it due to uncertainty about how future changes in the native code and wrapper might be affected. Specifically, I wasn't sure if we need to initialize the configuration first and then update it, or if we can do both simultaneously.

Looking at this again, I realize this method was mainly added for wrappers, where we have more control over initializing the module. I'll switch to using the constructor, as it seems like a better option. If we encounter limitations that require early configuration initialization, we can reintroduce this method to overcome the problem.